ui/app/src/state/epics/pathNavigator.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (C) 2007-2020 Crafter Software Corporation. All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as published by
+ * the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Epic, ofType, StateObservable } from 'redux-observable';
+import { map, mergeMap, withLatestFrom } from 'rxjs/operators';
+import { catchAjaxError } from '../../utils/ajax';
+import {
+  pathNavigatorFetchPath,
+  pathNavigatorFetchPathComplete,
+  pathNavigatorFetchPathFailed,
+  pathNavigatorInit,
+  pathNavigatorSetCurrentPath
+} from '../reducers/pathNavigator';
+import { getChildrenByPath } from '../../services/content';
+import GlobalState from '../../models/GlobalState';
+import { NEVER } from 'rxjs';
+
+export default [
+  (action$, state$: StateObservable<GlobalState>) =>
+    action$.pipe(
+      ofType(pathNavigatorFetchPath.type, pathNavigatorSetCurrentPath.type, pathNavigatorInit.type),
+      withLatestFrom(state$),
+      mergeMap(([{ type, payload: { id, path } }, state]) => {
+          if (pathNavigatorInit.type === type && localStorage.getItem('pathNavigator')) {
+            return NEVER;
```

Comment:
Why listen to _Init_ if you don't want to do anything with it?

Reply:
because pathNavigatorInit.type also needs to call getChildrenByPath, but only when there are not localStorage.getItem('pathNavigator').